src/ai/ai.cpp

```diff
@@ -4209,49 +4209,121 @@ void gather_to_battle(sys::state& state, dcon::nation_id n, dcon::province_id p)
 	}
 }
 
-float estimate_army_strength(sys::state& state, dcon::army_id a) {
+float estimate_balanced_composition_factor(sys::state& state, dcon::army_id a) {
```

Comment:
Did you take a performance measurement for this? You have turned what was meant to be a cheap function call into a quite complex one?